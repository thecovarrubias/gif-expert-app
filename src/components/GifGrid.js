import React from 'react';
import PropTypes from 'prop-types';
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';

const GifGrid = ({ category }) => {

    const { data: images, loading } = useFetchGifs( category );

    return (
        <section className="container">
            <h2 className="animate__animated animate__fadeIn lead mb-4">{ category.toUpperCase() }</h2>
            { loading && (
                <div className="spinner-grow text-secondary animate__animated animate__flash" role="status">
                    <span className="sr-only">Cargando...</span>
                </div>)
            }

            <div className="card-columns mb-5">
                {
                    images.map((img) => (
                        <GifGridItem 
                            key={ img.id } 
                            { ...img }
                        />
                    ))
                }
            </div>
        </section>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}

export default GifGrid;