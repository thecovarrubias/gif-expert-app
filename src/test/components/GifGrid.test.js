import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />', () => {

    const category = 'One Punch Man';

    test('Debe de mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                title: 'Cualquier cosa 1',
                url: 'https://localhost/cualquier/cosa_1.jpg'
            },
            {
                id: 'DEF',
                title: 'Cualquier cosa 2',
                url: 'https://localhost/cualquier/cosa_2.jpg'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={ category } />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    });

});