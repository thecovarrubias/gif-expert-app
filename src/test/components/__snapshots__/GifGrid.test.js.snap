// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    One Punch Man
  </h2>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    One Punch Man
  </h2>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa 1"
      url="https://localhost/cualquier/cosa_1.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Cualquier cosa 2"
      url="https://localhost/cualquier/cosa_2.jpg"
    />
  </div>
</Fragment>
`;
